webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/HttpsDataVault.scala

```diff
@@ -0,0 +1,104 @@
+package com.scalableminds.webknossos.datastore.datavault
+
+import com.scalableminds.webknossos.datastore.storage.{
+  FileSystemCredential,
+  HttpBasicAuthCredential,
+  LegacyFileSystemCredential,
+  RemoteSourceDescriptor
+}
+
+import java.net.URI
+import scala.concurrent.duration.DurationInt
+import sttp.client3._
+import sttp.model.Uri
+
+class HttpsDataVault(credential: Option[FileSystemCredential]) extends DataVault {
+
+  private val connectionTimeout = 5 seconds
+  private val readTimeout = 1 minute
+  private lazy val backend = HttpClientSyncBackend(options = SttpBackendOptions.connectionTimeout(connectionTimeout))
+
+  def getBasicAuthCredential: Option[HttpBasicAuthCredential] =
+    credential match {
+      case Some(c) => {
+        c match {
+          case h: HttpBasicAuthCredential    => Some(h)
+          case l: LegacyFileSystemCredential => Some(l.toBasicAuth)
+          case _                             => None
+        }
+      }
+      case None => None
+    }
+
+  private def authenticatedRequest() =
+    getBasicAuthCredential.map { credential =>
+      basicRequest.auth.basic(credential.username, credential.password)
+    }.getOrElse(
+        basicRequest
+      )
+      .readTimeout(readTimeout)
+
+  private def headRequest(uri: URI): Request[Either[String, String], Any] =
+    authenticatedRequest().head(Uri(uri))
+
+  private val headerInfoCache = scala.collection.mutable.Map[URI, (Boolean, Int)]()
+
+  private def getHeaderInformation(uri: URI) =
+    headerInfoCache.get(uri) match {
+      case Some((b, i)) => (b, i)
+      case None => {
+        val response: Identity[Response[Either[String, String]]] = backend.send(headRequest(uri))
+        val acceptsPartialRequests = response.headers.find(_.is("Accept-Ranges")).exists(_.value == "bytes")
+        val dataSize = response.headers.find(_.is("Content-Length")).map(_.value.toInt).getOrElse(0)
```

Comment:
```suggestion
        val dataSize = response.headers.find(_.is("Content-Length")).map(_.value.toLong).getOrElse(0L)
```
I guess files on the server may be bigger than 4GB.

Reply:
Hmm, this means we should also probably take `NumericRange[Long]` instead of the Int-specific `Range` in the Vault signatures, so that we can read chunks that are beyond the 4GB limit.